components/places/src/sync/mod.rs

```diff
@@ -0,0 +1,5 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
I feel like we might be overdoing it with module nesting in the `places` crate... Maybe just a `places::history_sync` module, instead of `places::sync::history`.

Also, I think importing the module as `history` without a rename to `history_sync` would always be confusing.